UI/js/pages/auth.js

```diff
@@ -0,0 +1,177 @@
+let loginbutton = document.getElementById("login-button");
+let login_email,
+  login_password = "";
+
+// add the login event handlers
+
+// login_email event handler
+document.getElementById("login_email").oninput = function(event) {
+  login_email = event.target.value;
+  validateLogin();
+};
+
+// login_password event handler
+document.getElementById("login_password").oninput = function(event) {
+  login_password = event.target.value;
+  validateLogin();
+};
+// this function disables the login button
+const disableloginbutton = () => loginbutton.setAttribute("disabled", true);
+
+// validate login function
+const validateLogin = () => {
+  // checking if both emptyness of values
+  if (
+    !doesFieldContainValidValue(login_email) &&
+    !doesFieldContainValidValue(login_password)
+  ) {
+    disableloginbutton();
+    loginbutton.setAttribute("value", "ENTER BOTH EMAIL & PASSWORD TO LOGIN");
+  } // checks if email is the only blank field
+  else if (!doesFieldContainValidValue(login_email)) {
+    disableloginbutton();
+    loginbutton.setAttribute("value", "ENTER EMAIL TO LOGIN");
+  } // checks if the password field is the only blank field
+  else if (!doesFieldContainValidValue(login_password)) {
+    disableloginbutton();
+    loginbutton.setAttribute("value", "ENTER PASSWORD TO LOGIN");
+  } // check if the email is in the right format
+  else if (!emailValidation(login_email)) {
+    disableloginbutton();
+    loginbutton.setAttribute("value", "ENTER A VALID EMAIL TO LOGIN");
+  } // if the password is less than 6 chars , dont validate
+  else if (login_password.length < 6) {
+    disableloginbutton();
+    loginbutton.setAttribute(
+      "value",
+      "THE PASSWORD MUST BE LONGER THAN 6 CHARACTERS"
+    );
+  } else {
+    loginbutton.setAttribute("value", "LOGIN");
+    loginbutton.removeAttribute("disabled");
+  }
+};
+
+validateLogin();
+
+// signup validation
+let registerbutton = document.getElementById("register-button");
+let firstname,
+  lastname,
+  othername,
+  phoneNumber,
+  passportUrl,
+  email,
+  password,
+  confirm = "";
+
+document.getElementById("firstname").oninput = function(event) {
```

Comment:
Since you are doing the same thing for every field here, it's good to create a single method the pass in the filed name to validated, this will ensure your code is `DRY`
i.e
```
const validateField = (fieldName) => {
    document.getElementById(fieldName).oninput = function(event) {
        fieldName = event.target.value;
        validateSignup();
    };
}
```
I don't know how you'd hock this fuction to be called on on every inputfield, but it would be a nice way to reduce the duplicated code.

Reply:
This can be an interesting challenge. I will think of a way in which I can make the event handler reusable